metagraph/src/cli/parse_sequences.hpp

```diff
@@ -54,7 +54,7 @@ void parse_sequences(const std::string &file,
                 [&](std::string_view, uint32_t count) {
                     count_hist[count] += (1 + config.forward_and_reverse);
                 },
-                !config.canonical && !config.forward_and_reverse
+                config.graph_mode != graph::DeBruijnGraph::CANONICAL && !config.forward_and_reverse
```

Comment:
if the graph mode is set to PRIMARY and a non-canonicalized/non-primary sequence set is passed, how does this affect the counting? will config.forward_and_reverse be true in this case?

Reply:
This is complicated. I tried to take into account all scenarios.
I'll try to simplify this now and forbid some combinations that don't make sense in applications, to make it more straightforward.